api/hasura/actions/upload_circle_logo.ts

```diff
@@ -0,0 +1,77 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { authCircleAdminMiddleware } from '../../../api-lib/circleAdmin';
+import { gql } from '../../../api-lib/Gql';
+import { ErrorResponse } from '../../../api-lib/HttpError';
+import { resizeCircleLogo } from '../../../api-lib/images';
+import { ImageUpdater } from '../../../api-lib/ImageUpdater';
+import {
+  composeHasuraActionRequestBodyWithSession,
+  HasuraUserSessionVariables,
+  uploadCircleImageInput,
+} from '../../../src/lib/zod';
+
+const handler = async function (req: VercelRequest, res: VercelResponse) {
+  const {
+    input: { object: input },
+    // session_variables: sessionVariables,
+  } = composeHasuraActionRequestBodyWithSession(
+    uploadCircleImageInput,
+    HasuraUserSessionVariables
+  ).parse(req.body);
+
+  try {
+    const previousLogo = await getPreviousLogo(input.circle_id);
+
+    const updater = new ImageUpdater<{ id: number }>(
```

Comment:
The more I look at the `ImageUpdater` class the more i appreciate it. 

I might be wrong about this, but you shouldn't need to explicitly add the <`T`> when instantiating the class. Typescript should be able to infer this from the mutation function you pass in. If you want to explicitly pass it in, that's cool, as long as typechecking fails when it doesn't match the actual return.

Reply:
I like being explicit 🤷

Reply:
I actually think the generic aspect of it might be unnecessary altogether since it always returns the same thing in the current impl, I was just geeking out a bit and trying to be flexible. There may be a version in future where we want to return { id: number, profile: x} or something for example